### PR TITLE
Enhance Version Compatibility in helmrelease-plugins.yaml for Advanced Plugin Functionality

### PR DESCRIPTION
**Pull request by jash2105**
Restructures the plugin configuration in `helmrelease-plugins.yaml` to improve modularity and streamline management. Configurations are organized to enhance accessibility for each plugin, simplifying future updates. The version constraint has been updated to `'>=2.4.0 <2.5.0'` to incorporate the latest stable features and ensure compatibility within the 2.x series.

**Changes**:
- Modular restructure of plugin configuration.
- Version range refined to `>=2.4.0 <2.5.0` for optimal stability with recent Kubecost updates.

### CODE REVIEW

**Comment by jash2105** [comment]
@jessegoodier , let me know if something needs to be changed!

File: fluxcd/federated-etl-flux/plugins/README.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+- **Automated Cluster Turndown**
+- **Cluster Right-Sizing Recommendations**
+- **Container Request Right-Sizing (RRS) Recommendations**
+- **Kubescaler (Alpha Mode)**
+- **Carbon Costs**
+
+
+## Enabling the Cluster Controller
+
+To use the following plugins:
```
Comment (jessegoodier) [inline fluxcd/federated-etl-flux/plugins/README.md:15]:
```suggestion
To use the following Kubecost Actions:
```

File: fluxcd/federated-etl-flux/plugins/README.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
+# Kubecost Additional Plugins
```
Comment (jessegoodier) [inline fluxcd/federated-etl-flux/plugins/README.md:1]:
```suggestion
# Kubecost Actions
```

Carbon costs is just an optional feature. everything else is considered Kubecost actions.

**Comment by jash2105** [comment]
Made the changes @jessegoodier

**Pull request merged**